src/AppInstallerRepositoryCore/Microsoft/PreIndexedPackageSourceFactory.cpp

```diff
@@ -21,6 +21,22 @@ namespace AppInstaller::Repository::Microsoft
         // TODO: This being hard coded to force using the Public directory name is not ideal.
         static constexpr std::string_view s_PreIndexedPackageSourceFactory_IndexFilePath = "Public\\index.db"sv;
 
+        struct PersistedIndexPackage
+        {
+            PersistedIndexPackage(const std::filesystem::path& path)
+            {
+                m_file = Utility::ManagedFile::OpenWriteLockedFile(path, 0);
+            }
+
+            bool ValidateMsixTrustInfo(bool checkMicrosoftOrigin) const
+            {
+                return Msix::ValidateMsixTrustInfo(m_file.GetFilePath(), checkMicrosoftOrigin);

```

Comment:
The reason to move to this model was that the *only* way to call `ValidateMsixTrustInfo` would be to create a `PersistedIndexPackage`.  If you leave the free function `ValidateMsixTrustInfo` around, someone can still use it without the file lock.